test/OpenTelemetry.Instrumentation.Http.Tests/HttpClientTests.Basic.cs

```diff
@@ -317,6 +326,28 @@ public async Task HttpClientInstrumentationBacksOffIfAlreadyInstrumented()
             Assert.Equal(4, processor.Invocations.Count); // SetParentProvider/OnShutdown/Dispose/OnStart called.
         }
 
+        [Fact]
+        public async Task HttpClientRedirectTest()
+        {
+            var processor = new Mock<BaseProcessor<Activity>>();
+            using (Sdk.CreateTracerProviderBuilder()
+                       .AddHttpClientInstrumentation()
+                       .AddProcessor(processor.Object)
+                       .Build())
+            {
+                using var c = new HttpClient();
+                await c.GetAsync($"{this.url}redirect");
+            }
+
+            Assert.Equal(7, processor.Invocations.Count); // SetParentProvider/OnStart/OnEnd/OnStart/OnEnd/OnShutdown/Dispose called.
```

Comment:
Would be nice to get this replaced with InMemoryExporter..

Reply:
Can do a follow up as this needs to be changed in other tests as well.